Feature/room service

Bagriell opened the pull request.

Bagriell committed:
entities and getall

Bagriell committed:
room service method

manon-violeau commented on back/src/services/workshop/rooms.ts:
```diff
@@ -1,43 +1,93 @@
+            }
+        ]
+    })
+    return rooms.concat(rooms_player_in)
```
Pas besoin de filtrer sur is_private, le user peut voir les rooms private même s'il n'est pas dedans. Il faut plutot filtrer sur : 
is_published : true,
is_published : false and gm.id = user.id,

manon-violeau commented on back/src/services/workshop/rooms.ts:
```diff
@@ -1,43 +1,93 @@
-    // return rooms
+    const user = await UserRepository.findOneByOrFail({ username: username })
+    const rooms_pub = await RoomRepository.findOneByOrFail({ id: room_id, is_private: false })
+    return rooms_pub
```
Pareil ici, le user peut accéder à une room private comme une room public. Il ne peut pas accéder à une room is_published: false and gm.id != user.id

Bagriell replied:
done

manon-violeau commented on back/src/database/migrations/1679057873701-roomUpdateGameRelationMigrations.ts:
```diff
@@ -0,0 +1,35 @@
+        await queryRunner.query(`
+            ALTER TABLE "room"
+            ALTER COLUMN "gmId"
+            SET NOT NULL
```
il faut ajouter la même contrainte sur gameId SET NOT NULL

Bagriell replied:
done

manon-violeau commented on back/src/services/workshop/rooms.ts:
```diff
@@ -1,43 +1,93 @@
+    // links resources
+    const new_player = await PLayerRepository.save({ user: user })
+    room_to_join.game.players.push(new_player)
+    return await RoomRepository.save(room_to_join)
```
manque le password

Bagriell replied:
done

Bagriell committed:
room related entities and service corrections

- unique constraint user-game
- non nullable table Rooms gm & game
- related  migrations files
- swagger routes
- room service checks and corrections

manon-violeau commented on back/src/database/entities/public/Players.ts:
```diff
@@ -21,5 +22,5 @@ export class Player {
     game: Game;
```
on pourrai ajouter un nullable false sur user et game (pas obligatoire en vrai)

manon-violeau reviewed (approved):
Top !
